nsxt/resource_nsxt_compute_manager.go

```diff
@@ -0,0 +1,531 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/bindings"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/data"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/fabric"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt-mp/nsx/model"
+)
+
+var accessLevelForOidcValues = []string{
+	model.ComputeManager_ACCESS_LEVEL_FOR_OIDC_FULL,
+	model.ComputeManager_ACCESS_LEVEL_FOR_OIDC_LIMITED,
+}
+
+func resourceNsxtComputeManager() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtComputeManagerCreate,
+		Read:   resourceNsxtComputeManagerRead,
+		Update: resourceNsxtComputeManagerUpdate,
+		Delete: resourceNsxtComputeManagerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"revision":     getRevisionSchema(),
+			"description":  getDescriptionSchema(),
+			"display_name": getDisplayNameSchema(),
+			"tag":          getTagsSchema(),
+			"access_level_for_oidc": {
+				Type:         schema.TypeString,
+				Description:  "Specifies access level to NSX from the compute manager",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice(accessLevelForOidcValues, false),
+				Default:      model.ComputeManager_ACCESS_LEVEL_FOR_OIDC_FULL,
+			},
+			"create_service_account": {
+				Type:        schema.TypeBool,
+				Description: "Specifies whether service account is created or not on compute manager",
+				Optional:    true,
+				Default:     false,
+			},
+			"credential": {
+				Type:        schema.TypeList,
+				Description: "Login credentials for the compute manager",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"saml_login_credential": {
```

Comment:
I would suggest to remove `credential` suffix here since its already present in parent name